### DISCUSSION TITLE
AWS SSO Login PowerShell

### DISCUSSION THREAD

**Discussion by edwio** [aws/aws-tools-for-powershell#306] [General]
Due to security policy, connecting to AWS programmatically with PowerShell using a Access/Secret Key is not allowed at our organization anymore, and need to be replaced with AWS SSO. But I didn't find any clear documention about how to actually connect to AWS with PowerShell, using AWS SSO. On AWS page for PowerShell Cmdlets I can see a cmdlet named: Close-SSOSession, which is very straight forward about what is does. but which cmdlet is used to open the session in the first place? I would expect something like: Open-SSOSession. I also need to note, that we are not allowed to install the AWS CLI V2 on our Windows servers, As only the AWS PoweShel Module is approved to use by developers.

**Comment by ashishdhingra** [comment]
@edwio [Single sign-on (SSO) with the AWS SDK for .NET](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/sso.html) provides guidance on how to load SSO credentials using .NET application only. As of now, there doesn't appear to a CmdLet that exposes this functionality. May be try re-writing the .NET logic in PowerShell script could help.

**Reply by ashishdhingra** [reply]
@edwio We have a new feature request https://github.com/aws/aws-tools-for-powershell/issues/332. This is queued in team's backlog.

@github-actions proposed-answer
